test/Constraints/closures.swift

```diff
@@ -894,8 +894,12 @@ do {
   }
 
   func foo(_ arr: [Int]) {
+    // FIXME: This behavior related to tuple splat being allowed
+    //        in conversion between a single dependent member
+    //        parameter and empty parameter functions e.g.
+    //        () -> Void `convertable to` (T.V) -> Void.
     _ = S(arr, id: \.self_) {
-      // expected-error@-1 {{contextual type for closure argument list expects 1 argument, which cannot be implicitly ignored}} {{30-30=_ in }}
+      // expected-error@-1 {{type '_' has no member 'self_'}}
```

Comment:
I think fixing this completely will require not re-typechecking at all. You're replacing the dependent member type with UnresolvedType here.

Reply:
Actually this error is just unrelated :( `C.Element` is resolved properly to `Int` later on (on the first solve) so we end up comparing `()` to `Int` and failing. I'm trying to think of a way to fix this in `salvage` mode. Maybe we shouldn't attempt splat at all, and let `fixMissingArgument` and other fixes do their thing, I'll need to add fixes for extraneous and destructuring first though...

Reply:
Can we try both, build a disjunction with splat and without splat and see what fixes each possibility produces?

Reply:
I was thinking about trying to replace argument with type variable and then add conversion constraint to it with original type, this would act similar to disjunction because potential bindings would get both parameter and argument type.